UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -0,0 +1,91 @@
+        }
+        public async Task<CreateBatchResponse> CreateBatch()
+        {
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            var jwtSecurityToken = new JwtSecurityToken(accessToken);
+            var oid = jwtSecurityToken.Claims.FirstOrDefault(m => m.Type == "oid").Value;
+            var uri = $"/batch";
+
+            CreateBatchRequest createBatchRequest = new CreateBatchRequest
+            {
+                BusinessUnit = fileShareServiceConfig.Value.BusinessUnit,
+                Attributes = new List<KeyValuePair<string, string>>()
+                {
+                    new KeyValuePair<string, string>("Exchange Set Type", "Update"),
+                    new KeyValuePair<string, string>("Media Type", "Zip"),
+                    new KeyValuePair<string, string>("Product Type", "AVCS")
+                },
+                ExpiryDate = DateTime.UtcNow.AddDays(1).ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture),
+                Acl = new Acl()
+                {
+                    ReadUsers = new List<string>() { oid }
+                }
+            };
+            batchExpiryDateTime =createBatchRequest.ExpiryDate;
+
+            string payloadJson = JsonConvert.SerializeObject(createBatchRequest);
+
+            var httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            CreateBatchResponse createBatchResponse = await CreateBatchResponse(httpResponse);
+            return createBatchResponse;                               
+        }
+
+        private async Task<CreateBatchResponse> CreateBatchResponse(HttpResponseMessage httpResponse)
+        {
+            var createBatchResponse = new CreateBatchResponse();
+            var body = await httpResponse.Content.ReadAsStringAsync();
+            if (httpResponse.StatusCode != HttpStatusCode.Created)
+            {
+                logger.LogError(EventIds.FSSCreateBatchNonOkResponse.ToEventId(), $"File share service create batch endpoint responded with {httpResponse.StatusCode} and message {body}");
+                createBatchResponse.ResponseCode = httpResponse.StatusCode;
+                createBatchResponse.ResponseBody = null;
+            }
+            else
+            {
+                createBatchResponse.ResponseCode = httpResponse.StatusCode;
+                createBatchResponse.ResponseBody = JsonConvert.DeserializeObject<CreateBatchResponseModel>(body);
+                createBatchResponse.ResponseBody.BatchStatusUri =$"{fileShareServiceConfig.Value.BaseUrl}/batch/{createBatchResponse.ResponseBody.BatchId}";
+                createBatchResponse.ResponseBody.BatchExpiryDateTime = batchExpiryDateTime;
+                createBatchResponse.ResponseBody.ExchangeSetFileUri =$"{createBatchResponse.ResponseBody.BatchStatusUri}/files/{fileShareServiceConfig.Value.ExchangeSetFileName}";
```

Comment:
Is `ExchangeSetFileUri` not a fully qualified url? We didn't use FSS base url for it, like we did for BatchStatusUri .

Reply:
For ExchangeSetFileUri, the initial part of the url(fssbaseurl/batch/batchid) is same as BatchStatusUri. Hence we have reused value of variable createBatchResponse.ResponseBody.BatchStatusUri instead of setting FSS baseurl once again.


createBatchResponse.ResponseBody.BatchStatusUri =$"fileShareServiceConfig.Value.BaseUrl}/batch/{createBatchResponse.ResponseBody.BatchId}";
createBatchResponse.ResponseBody.ExchangeSetFileUri =$"{createBatchResponse.ResponseBody.BatchStatusUri}/files/{fileShareServiceConfig.Value.ExchangeSetFileName}";

Reply:
ok, thanks.